### PR TITLE
Adjusted wording in pull request guidelines

### LINKED ISSUE #22932: [Doc]: small edits to the Pull request guidelines
### Documentation Link

https://matplotlib.org/devdocs/devel/coding_guide.html

### Problem

- In the first note, second bullet, after `Nevertheless,` and before `please try to follow the guidelines...` 
    - Add the word `please` 
- Change the PR in the following subtitles to say `pull request`. So that all the titles and subtitles on the page consistently use the full term `pull request(s)` and not he shorthand `PR`
    - `Summary for PR authors`
    - `Summary for PR reviewers`

### Suggested improvement

- In the first note, second bullet, after `Nevertheless, please try to follow the guidelines...`
- Change the PR in the following subtitles to say `pull request`
    - `Summary for pull request authors`
    - `Summary for pull request reviewers`

### PR DESCRIPTION
**Pull request by andrew-fennell**
## PR Summary
Adjusted pull request guidelines titles from PR to "pull request" to match the title in the section. 

Requested by @noatamir in #22932

Closes #22932

## PR Checklist

**Tests and Styling**
- [N/A] Has pytest style unit tests (and `pytest` passes).
- [N/A] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: doc/devel/coding_guide.rst [hunk 1]
```diff
@@ -16,15 +16,15 @@ Pull request guidelines
 Pull requests (PRs) are the mechanism for contributing to Matplotlibs code and
 documentation.
 
-Summary for PR authors
+Summary for pull request authors
 ======================
```
Comment (tacaswell) [inline doc/devel/coding_guide.rst:20]:
```suggestion
================================
```

File: doc/devel/coding_guide.rst [hunk 2]
```diff
@@ -54,7 +54,7 @@ When making a PR, pay attention to:
 
 See also :ref:`contributing` for how to make a PR.
 
-Summary for PR reviewers
+Summary for pull request reviewers
 ========================
```
Comment (tacaswell) [inline doc/devel/coding_guide.rst:58]:
```suggestion
==================================
```

**Comment by tacaswell** [comment]
We our doc build set up to fail on warnings (as that is the only way to keep this amount of rst in any sort of reasonable shape).  I took the liberty of fixing the underline lengths.

**Comment by andrew-fennell** [comment]
@tacaswell Is that what caused the tests to fail?

**Review by tacaswell — APPROVED** [review]

**Comment by tacaswell** [comment]
The exact reason is these lines:

```
reading sources... [100%] users/resources/index

/home/circleci/project/doc/devel/coding_guide.rst:20: WARNING: Title underline too short.

Summary for pull request authors
======================
/home/circleci/project/doc/devel/coding_guide.rst:58: WARNING: Title underline too short.

Summary for pull request reviewers
========================
/home/circleci/project/doc/devel/coding_guide.rst:58: WARNING: Title underline too short.
```

https://app.circleci.com/pipelines/github/matplotlib/matplotlib/16058/workflows/c75b8fdc-0e51-4668-b16d-69afb4f754f8/jobs/68030?invite=true#step-113-832

**Comment by tacaswell** [comment]
@andrew-fennell A bit of GH magic if you put

```
Closes #XYZ 
 ```

in a commit message body github will link the PR to the issue referenced.  This will both put a note in the comment thread on the issue it will automatically close the issue when the PR is merged.  It is a little thing, but it really helps keep the issues tidy!

**Review by oscargus — APPROVED** [review]

**Comment by oscargus** [comment]
@meeseeksmachine backport to v3.5.x

**Comment by oscargus** [comment]
@meeseeksdev backport to v3.5.x

**Pull request merged**